### PR TITLE
=doc Consolidate route directives docs #1624

### PR DESCRIPTION
**Pull request by jlprat**
Refs #1624
Convert directory to symlink
Add example on `redirect`

### CODE REVIEW

File: docs/src/test/java/docs/http/javadsl/server/directives/RouteDirectivesExamplesTest.java [hunk 1]
```diff
@@ -3,22 +3,35 @@
...
 
 public class RouteDirectivesExamplesTest extends JUnitRouteTest {
 
+  @Test
+  public void testShowRedirectOnCompleteWithTerms() {
+    final StatusCode redirectionType = StatusCodes.FOUND;
+    final Uri uri = Uri.create("http://akka.io");
+    final Function<RequestContext, CompletionStage<RouteResult>> route = rc ->
+            //#red-impl
+            rc.completeWith(HttpResponse.create()
```
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/server/directives/RouteDirectivesExamplesTest.java]: Quite an artificial example, but it is just to show in Java code how `redirect` is defined on `completeWith` terms

File: docs/src/main/paradox/scala/http/routing-dsl/directives/route-directives/complete.md [hunk 2]
```diff
@@ -25,6 +28,8 @@ with the respective @unidoc[HttpResponse] instance.
 Completing the request will send the response "back up" the route structure where all the logic runs that wrapping
 directives have potentially chained into the @ref[RouteResult](../../routes.md#routeresult) future transformation chain.
 
+@java[Please note that the `complete` directive has multiple variants, like the ones showed in the examples.]
```
Comment (jrudolph) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/route-directives/complete.md]: "shown"?

Comment (jlprat) [reply docs/src/main/paradox/scala/http/routing-dsl/directives/route-directives/complete.md]: indeed

File: docs/src/test/java/docs/http/javadsl/server/directives/RouteDirectivesExamplesTest.java (hunk 1 above)
Comment (jrudolph) [reply docs/src/test/java/docs/http/javadsl/server/directives/RouteDirectivesExamplesTest.java]: Yep, why not, if that's consistent with the Scala side.

**Pull request merged**